### PR TITLE
Chatbot Attachments

### LINKED ISSUE #121: Chat Bot Epic
This Epic has tickets related to the chat bot functionality for IoGT. Significant work for the functionality has been done by @geoo89 and he should be the contact point for any questions/queries.

Here is a list of features that need to be done as a part of this epic:
- Review `messaging.models` - Use managers where applicable
- Refactor messaging app to be more django-esque
- Replace `urllib.request` with `requests` package
- Update existing test cases to use Django Test framework (contrary to DjangoTestPlus)
- Add Model Factories for `messaging` Test Cases
- Complete skipped test cases
- Use [rapidpro-python](https://github.com/rapidpro/rapidpro-python) or write custom client for communication with RapidPro server
- Decide on async/sync communication with RapidPro server
- Refactor the RapidPro webhook to use DRF
- Refactor Test Cases
- Stitch longer messages to single message on the IoGT website
- Make quick replies "clickable" so the user doesn't have to type them
- Preview image links to show images (not a file)
- Give the editor an option to start a chat with the chosen bot based on a button
- Add Chatbot settings to Wagtail Admin

Additionally, once the above functionality is working, we'd have to look into the following problems:
- The communication from IoGT to RapidPro isn't secure because there is no authentication layer. We need to figure out a way to make this communication secure.
- Test the integration on a newer version of IoGT
- Figure out a way to auto update the screen or intimate the user that their latest messages will be shown only when they update the page

### PR DESCRIPTION
**Pull request by ehmadzubair**
This PR has the follwing changes:
- Parse Attachments from rapidpro messages text
- Download attachments from rapidpro messages (if they don't already exist)
- Show Attachment previews (if they are valid images)
- Serve internal (downloaded) media
- Basic test case for message parsing

Fixes #121

### CODE REVIEW

File: messaging/models.py [hunk 1]
```diff
@@ -74,8 +79,34 @@ class Message(models.Model):
...
     class Meta:
         ordering = ("sent_at",)
 
     def get_absolute_url(self):
         return self.thread.get_absolute_url()
+
+
+class Attachment(TimeStampedModel):
+    external_link = models.URLField(null=False, blank=False)
+    downloaded_file = models.FileField(null=False, blank=False)
```
Comment (sheralim012) [inline messaging/models.py:101]: `null=False, blank=False`is by default, no need to add explicitly

File: messaging/models.py [hunk 2]
```diff
@@ -74,8 +79,34 @@ class Message(models.Model):
...
+
+    def download_external_file(self):
+        response = requests.get(self.external_link, allow_redirects=True)
+
+        if response.status_code == status.HTTP_200_OK:
+            image_file = File(BytesIO(response.content), name=self.external_link.split('/')[-1])
```
Comment (sheralim012) [inline messaging/models.py:107]: attachments can be of other types so update name

File: messaging/models.py [hunk 3]
```diff
@@ -74,8 +79,34 @@ class Message(models.Model):
...
+            self.downloaded_file = image_file
+            self.save()
```
Comment (sheralim012) [inline messaging/models.py:109]: use `updated_fields` kwarg

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Comment by ehmadzubair** [comment]
@geoo89 this is ready to merge

**Review by sheralim012 — APPROVED** [review]

**Comment by geoo89** [comment]
Thanks, this looks great! I noticed a few minor things, but I'll create a new issue and merge this in.

**Pull request merged**